tianshou/data/buffer.py

```diff
@@ -141,6 +147,14 @@ def __getattr__(self, key: str) -> Union['Batch', Any]:
         """Return self.key"""
         return self._meta.__dict__[key]
 
+    def __setstate__(self, state):
```

Comment:
Are you sure it is not working out-of-the-box ?

Reply:
What do you mean?

Reply:
usually, it is not necessary to define this can of interface for Pickle to work, unless you need to do some fancy stuff in it. And from what I'm saying, what you are doing looks like the default behaviour, but I may be wrong.

Reply:
> I did so because I think it is not the best to re-initialize the buffer, because there are so many metavar to be restored (unlike Batch).

Yes and your are right. But this is what pickle should be doing by default. It should not call the constructor (`__init__`), not even `__new__` if I remember well.

Reply:
> usually, it is not necessary to define this can of interface for Pickle to work, unless you need to do some fancy stuff in it. And from what I'm saying, what you are doing looks like the default behavior, but I may be wrong.

Yes, you are right. I fix the existing `__getattr__` to exclude `self._meta["__getstate__"]` happen and it works without `__getstate__` and `__setstate__`.

Reply:
Ok so it should be mentioned in the doc that user keys must not start with `__` (which means hidden in Python anyway so it makes total sense)